### ISSUE TITLE
can not connect server without password

### ISSUE THREAD

**Issue by kuyagic** [kopia/kopia#902]
I start the server with tls and `--without-password` and `--insecure` flag but set the repo user, `--allow-repository-user` is not available in 0.8.0-beta3, so I omitted this flag.

and I connect the server ,server raised the error and exited.
```
server start --tls-cert-file /usr/local/etc/kopia/server.pem --tls-key-file /usr/local/etc/kopia/serve
r.key --without-password --insecure --address=0.0.0.0:51515 --file-log-level=error
```

client connect command and provide the repo's user password
```
kopia repository connect server --url=kopia://192.168.11.12:51515 --server-cert-fingerprint=ac860138
0682de892e31457b589f9eb1c60b7473a014c701f580a1a911a6b784
```

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0xc2ad5c]

goroutine 96 [running]:
github.com/kopia/kopia/internal/server.(*Server).authenticateGRPCSession(0xc0000be3c0, 0x183da08, 0xc00475aab0, 0x1850088, 0xc0001aa5a0, 0x1110301, 0x0)
        /home/runner/work/kopia/kopia/internal/server/grpc_session.go:59 +0x25c
github.com/kopia/kopia/internal/server.(*Server).Session(0xc0000be3c0, 0x1848e70, 0xc00479c8d0, 0x0, 0x0)
        /home/runner/work/kopia/kopia/internal/server/grpc_session.go:78 +0x125
github.com/kopia/kopia/internal/grpcapi._KopiaRepository_Session_Handler(0x12b23e0, 0xc0000be3c0, 0x1844e40, 0xc0000dacc0, 0x1f8bf10, 0xc004692d00)
        /home/runner/work/kopia/kopia/internal/grpcapi/repository_server_grpc.pb.go:93 +0xad
google.golang.org/grpc.(*Server).processStreamingRPC(0xc0046d8000, 0x184a238, 0xc0046e0900, 0xc004692d00, 0xc0046abf80, 0x1f406a0, 0x0, 0x0, 0x0)
        /home/runner/go/pkg/mod/google.golang.org/grpc@v1.36.0/server.go:1464 +0xcd8
google.golang.org/grpc.(*Server).handleStream(0xc0046d8000, 0x184a238, 0xc0046e0900, 0xc004692d00, 0x0)
        /home/runner/go/pkg/mod/google.golang.org/grpc@v1.36.0/server.go:1544 +0xca5
google.golang.org/grpc.(*Server).serveStreams.func1.2(0xc00475c9f0, 0xc0046d8000, 0x184a238, 0xc0046e0900, 0xc004692d00)
        /home/runner/go/pkg/mod/google.golang.org/grpc@v1.36.0/server.go:878 +0xab
created by google.golang.org/grpc.(*Server).serveStreams.func1
        /home/runner/go/pkg/mod/google.golang.org/grpc@v1.36.0/server.go:876 +0x1fd
```

**Comment by jkowalski** [comment]
Can you describe your scenario? You have clients connecting to the server and want them to not use any authentication? `--without-password` is meant for a local user connecting to a local server where security is not important.

**Comment by jkowalski** [comment]
Also, `--allow-repository-users` is now default in 0.8.0-beta3 and the flag has been removed.

**Comment by kuyagic** [comment]
The situation is all data in my LAN. I started a server in one of my Linux box to provide backup from other PCs.
And at the same time I want to view it via web UI to access the file without the basic authentication.
I have set the password via `kopia server user add` command, just want to a read-only web UI for viewing files backup to the server.

I think kopia can and need to check the server's security for which the web UI's password is not be set, and deny the connection from client side, but should not raise the error.

- The re-produce flow is below

simply start the server with 
```
kopia server start --tls-cert-file /usr/local/etc/kopia/server.pem --tls-key-file /usr/local/etc/kopia/serve
r.key --without-password --insecure --address=0.0.0.0:51515 --file-log-level=error
```

and connect the server with 
```
kopia repository connect server --url=kopia://192.168.11.12:51515 --server-cert-fingerprint=ac860138
0682de892e31457b589f9eb1c60b7473a014c701f580a1a911a6b784
```
